ouroboros-consensus/src/Ouroboros/Storage/VolatileDB/Util.hs

```diff
@@ -74,20 +75,32 @@ wrapFsError :: Monad m
 wrapFsError fsErr volDBErr action =
     tryVolDB fsErr volDBErr action >>= either (throwError volDBErr) return
 
--- Throws an error if one of the given file names does not parse.
+-- | Parses the 'FsPath' of each filename and zips them together.
+-- If any 'FsPath' does not parse, we return an error.
+parseAllFds :: forall blockId.
+               [FsPath]
+            -> Either (VolatileDBError blockId) [(FileId, FsPath)]
+parseAllFds = mapM go
+  where
+    go :: FsPath -> Either (VolatileDBError blockId) (FileId, FsPath)
+    go file = case parseFd file of
+      Nothing -> Left $ UnexpectedError $ ParserError $ InvalidFilename file
+      Just fd -> Right (fd, file)
```

Comment:
You could consider using `throwError` instead of `Left` and `return` instead of `Right`. Up to you.